### PR TITLE
feat: add storage netapp alpha

### PR DESCRIPTION
**Pull request by antleblanc**
| Question         | Answer
| ---------------- | ---
| Branch?          | `master`
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | MANAGER-5654
| License          | BSD 3-Clause

### :boom: Breaking Changes

600dd23 - feat: add storage netapp alpha

### :sparkles: Features

6591e67 - feat: add @ovh-ux/manager-storage module
a28ebcd - feat(sidebar.constant): add storage config

### :link: Related

- #3821
- #4420

### :flags: Translations

- [x] TRANS-31864

Signed-off-by: Benjamin Garnier <benjamin.garnier@ovhcloud.com>

/cc @benjamin-garnier-pro

### CODE REVIEW

File: packages/manager/modules/storage/src/storages.service.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
+export default class StoragesService {
+  /* @ngInject */
+  constructor($q, OvhHttp) {
+    this.$q = $q;
+    this.OvhHttp = OvhHttp;
+  }
+
+  getStorageNasha() {
+    return this.OvhHttp.get(`/dedicated/nasha/`, {
```
Comment (frenautvh) [inline packages/manager/modules/storage/src/storages.service.js:9]: i guess we should use iceberg for those 3 methods

File: packages/manager/modules/storage/src/netapp/netapp.controller.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    this.NetappService = NetappService;
+    this.CucRegionService = CucRegionService;
+    this.CORE_URLS = CORE_URLS;
+    this.User = User;
+  }
+
+  initGuides() {
+    this.guidesLoading = true;
+    return this.User.getUser()
+      .then((user) => {
```
Comment (frenautvh) [inline packages/manager/modules/storage/src/netapp/netapp.controller.js:14]: Environment.getUser() here

File: packages/manager/modules/storage/src/netapp/netapp.service.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  }
+
+  getService(netappId) {
+    return this.OvhHttp.get(`/storage/netapp/${netappId}`, {
+      rootPath: 'apiv6',
+    });
+  }
+
+  getShares(netappId) {
+    return this.OvhHttp.get(`/storage/netapp/${netappId}/share`, {
```
Comment (frenautvh) [inline packages/manager/modules/storage/src/netapp/netapp.service.js:15]: iceberg here

File: packages/manager/modules/storage/package.json [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    "directory": "packages/manager/modules/storages"
+  },
+  "license": "BSD-3-Clause",
+  "author": "OVH SAS",
+  "main": "src/index.js",
+  "dependencies": {
+    "lodash": "^4.17.15"
+  },
+  "peerDependencies": {
+    "@ovh-ux/manager-core": "^9.0.0",
```
Comment (frenautvh) [inline packages/manager/modules/storage/package.json:18]: cloud universe components is used but is missing in peer deps

Comment (antleblanc) [reply packages/manager/modules/storage/package.json:18]: Package `@ovh-ux/ng-ovh-cloud-universe-components` has been added as a peer dependency.

**Comment by antleblanc** [comment]
Rebasing on `master` branch and rewriting commit history.

**Comment by antleblanc** [comment]
Rebased successfully on `master` branch.

File: packages/manager/modules/server-sidebar/src/sidebar.constants.js [hunk 5]
```diff
@@ -132,13 +142,22 @@ export const NETWORKS_CONFIG = {
       regions: ['EU', 'CA'],
       searchKeys: ['NAS', 'NASHA', 'NAS-HA'],
     },
+    {
+      path: '/storage/netapp',
+      state: 'netapp',
+      stateParams: ['netappId'],
+      icon: 'ovh-font ovh-font-cloudnas',
+      app: [DEDICATED],
+      regions: ['EU', 'CA'],
```
Comment (marie-j) [inline packages/manager/modules/server-sidebar/src/sidebar.constants.js:151]: Shouldn't it be feature flipped ?

File: packages/manager/modules/storage/src/netapp/netapp.controller.js [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+      'deleting',
+    ]);
+  }
+
+  isError() {
+    return this.isStatusIn(['suspended', 'deleted']);
+  }
+
+  initGuides() {
+    const user = Environment.getUser();
```
Comment (JeremyDec) [inline packages/manager/modules/storage/src/netapp/netapp.controller.js:31]:
```suggestion
    const user = coreConfig.getUser();
```

File: packages/manager/modules/storage/src/storages.controller.js [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+          ),
+        }),
+        {},
+      ),
+    };
+  }
+
+  getSorting(property) {
+    return this.sort === property ? this.sortOrder.toLowerCase() : '';
+  }
```
Comment (JeremyDec) [inline packages/manager/modules/storage/src/storages.controller.js:41]: This logic could be stored in a utils file to avoid code duplication.

**Comment by JeremyDec** [comment]
rebase on `develop`

**Comment by JeremyDec** [comment]
rebase on `master`

**Comment by antleblanc** [comment]
:pray: Thank you everyone for the review you've made on this following Pull Request.

I will close this Pull Request in order to restart the module and app creation
via our up-to-date scaffold tool.

Be sure that we will take in account all latest comments :speech_balloon:.

cc @frenautvh  @marie-j  @JeremyDec  @benjamin-garnier-pro

**Pull request closed without merging**